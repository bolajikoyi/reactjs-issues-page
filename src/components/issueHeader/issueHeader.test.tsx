import React from 'react';
import { render, screen, cleanup} from '@testing-library/react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import IssueHeader from './issueHeader';
import userEvent from '@testing-library/user-event';

afterEach(cleanup);

let data = {
    isClosed: true,
    totalCount: 30,
    issueState: jest.fn()
}

describe('IssueHeader Component', ()=> {
    test('Expect componet to be created', async () => {
        expect(IssueHeader).toBeTruthy()
    });

test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<IssueHeader data={data} />, div);
});

test('function returns open', () => {
    const click = jest.fn(() => 'OPEN');
    render(<div data-testid='open' onClick={click}></div>);
    userEvent.click(screen.getByTestId('open'));
    expect(click).toBeCalledTimes(1);
})

test('function returns closed', () => {
    const click = jest.fn(() => 'CLOSED');
    render(<div data-testid='closed' onClick={click}></div>);
    userEvent.click(screen.getByTestId('closed'));
    expect(click).toBeCalledTimes(1);
})

test("Should Contain Text 'Author'", () => {
    render(<IssueHeader data={data} />);
    expect(screen.getByTestId('author')).toHaveTextContent('Author');
});

test("Should not contain the text 'Writer'", () => {
    render(<IssueHeader data={data} />);
    expect(screen.getByTestId('author')).not.toHaveTextContent('Writer');
});

test('matches snapshot', ()=> {
    const tree = renderer.create(<IssueHeader data={data}/>).toJSON();
    expect(tree).toMatchSnapshot();
})
})
