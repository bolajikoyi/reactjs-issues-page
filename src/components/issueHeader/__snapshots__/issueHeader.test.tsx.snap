// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueHeader Component matches snapshot 1`] = `
<div>
  <div
    className="IssueHeader"
  >
    <div
      className="header"
    >
      <span
        className="open"
        data-testid="open"
        onClick={[Function]}
      >
        <i
          className="far fa-dot-circle"
        />
         
        492 Open
      </span>
      <span
        className="closed active"
        data-testid="closed"
        onClick={[Function]}
      >
        <i
          className="fas fa-check"
        />
         
        30 Closed
      </span>
      <span
        className="author"
        data-testid="author"
      >
        Author 
        <i
          className="fas fa-caret-down"
        />
      </span>
      <span
        className="label"
      >
        Label 
        <i
          className="fas fa-caret-down"
        />
      </span>
      <span
        className="projects"
      >
        Projects 
        <i
          className="fas fa-caret-down"
        />
      </span>
      <span
        className="milestones"
      >
        Milestones 
        <i
          className="fas fa-caret-down"
        />
      </span>
      <span
        className="assignee"
      >
        Assignee 
        <i
          className="fas fa-caret-down"
        />
      </span>
      <span
        className="sort"
      >
        Sort 
        <i
          className="fas fa-caret-down"
        />
      </span>
    </div>
  </div>
</div>
`;
